fh-js-sdk.d.ts

```diff
@@ -0,0 +1,503 @@
+     * 
+     * @type PushConfig
+     */
+    export interface PushConfig {
+        alias:string;
+        categories:Array<string>;
+    }
+
+    /**
+     * Interface for the default error object used with the error callback in the SDK.
+     * 
+     * @type FailureCallbackError
+     */
+    export interface FailureCallbackError {
+        status:number;
+        message:string;
+        error:any;
+    }
+
+    /**
+     * Initializes the client SDKs.
+     * 
+     * @param {Object} options
+     * @param {Function} success - A callback function to be run on success.
+     * @param {Function} failure - A callback function to be run on failure.
+     * 
+     * @returns {Function}
+     */
+    export function init(options:any, success:(host:string) => void, failure:(error:any) => void);
+
+    /**
+     * Deprecated - Use $fh.cloud instead
+     * 
+     * @param {Object} options
+     * @param {Function} success - A callback function to be run on success.
+     * @param {Function} failure - A callback function to be run on failure.
+     * 
+     * @returns {String} - The Cloud Host URL
+     */
+    export function act(options:{}, 
+        success:(data:any, status:any, xhr:XMLHttpRequest) => void, 
+        failure:(message:string, error:FailureCallbackError) => void);
+
+    /**
+     * Authenticate and optionally authorise a user via access rights management.
+     * 
+     * @param {Object} options
+     * @param {Function} success - A callback function to be run on success.
+     * @param {Function} failure - A callback function to be run on failure.
+     */
+    export function auth(options:{policyId:string, clientToken:string, endRedirectUrl?:string, params?:any}, 
+        success:(data:any, status:any, xhr:XMLHttpRequest) => void, 
+        failure:(message:string, error:FailureCallbackError) => void);
+
+    /**
+     * Call any cloud URLs which have been defined in the Cloud App using AJAX.
+     * 
+     * @param {Object} options
+     * @param {Function} success - A callback function to be run on success.
+     * @param {Function} failure - A callback function to be run on failure.
+     */
+    export function cloud(options:{path:string, method?:string, contentType?:string, data?:any, timeout?:number}, 
+        success:(data:any, status:any, xhr:XMLHttpRequest) => void, 
+        failure:(message:string, error:FailureCallbackError) => void);
+
+    /**
+     * Key pair generation and data encryption and decryption.
+     * 
+     * @param {Object} options
+     * @param {String} options.act - One of 'hash'|'encrypt'|'decrypt'|'keygen'.
+     * @param {Object} options.params - Options for the chosen act.
+     * @param {Function} success - A callback function to be run on success.
+     * @param {Function} failure - A callback function to be run on failure.
+     */
+    export function sec(options:{act:"hash"|"encrypt"|"decrypt"|"keygen", params:any}, 
```

Comment:
I think `hash` is supported [here](https://github.com/feedhenry/fh-js-sdk/blob/master/src/modules/api_sec.js#L20).

Yeah, definitely!